pkg/controller/util.go

```diff
@@ -211,6 +211,7 @@ func CreateScratchPersistentVolumeClaim(client client.Client, pvc *v1.Persistent
 	scratchPvc := &v1.PersistentVolumeClaim{}
 	if err := client.Get(context.TODO(), types.NamespacedName{Name: scratchPvcSpec.Name, Namespace: pvc.Namespace}, scratchPvc); err != nil {
 		klog.Errorf("Unable to get scratch space pvc, %v\n", err)
+		return nil, err
```

Comment:
this is interesting, but I did actually catch this a few times while doing manual tests